File: app/src/main/java/com/dev/nicola/allweather/utils/WeatherUtils.java
package com.dev.nicola.allweather.utils;

import com.dev.nicola.allweather.R;

/**
 * Created by Nicola on 17/08/2016.
 */
public class WeatherUtils {

    public static int getWeatherIcon(String condition) {
        int icon;

        switch (condition) {
            case "1000":
            case "clear-day":
            case "32":
                icon = R.drawable.clear_day;
                break;
            case "clear-night":
            case "31":
                icon = R.drawable.clear_night;
                break;
            case "1003":
            case "partly-cloudy-day":
            case "30":
            case "28":
                icon = R.drawable.partly_cloudy_day;
                break;
            case "partly-cloudy-night":
            case "29":
            case "27":
                icon = R.drawable.partly_cloudy_night;
                break;
            case "1087":
            case "1009":
            case "1006":
            case "cloudy":
            case "26":
                icon = R.drawable.cloud;
                break;
            case "1195":
            case "1192":
            case "1189":
            case "1186":
            case "1183":
            case "1063":
            case "rain":
            case "11":
            case "12":
                icon = R.drawable.rain;
                break;
            case "1225":
            case "1222":
            case "1219":
            case "1216":
            case "1213":
            case "1210":
            case "1114":
            case "1066":
            case "snow":
            case "16":
                icon = R.drawable.snow;
                break;
            case "1072":
            case "1069":
            case "sleet":
            case "18":
                icon = R.drawable.sleet;
                break;
            case "1117":
            case "wind":
            case "24":
                icon = R.drawable.wind;
                break;
            case "1147":
            case "1135":
            case "1030":
            case "fog":
            case "20":
                icon = R.drawable.fog;
                break;
            case "1273":
            case "1276":
            case "1279":
            case "1282":
            case "4":
            case "37":
            case "38":
            case "39":
            case "47":
                icon = R.drawable.storm;
                break;
            default:
                icon = R.drawable.unknown;
                break;
        }
        return icon;
    }


    public static String getWindDirection(int degrees) {
        String direction;
        String cardinal[] = {"N", "NNE", "NE", "ENE", "E", "ESE", "SE", "SSE", "S", "SSW", "SW", "WSW", "W", "WNW", "NW", "NNW"};
        int n = (int) ((degrees / 22.5) + 0.5);
        direction = cardinal[n % cardinal.length];
        return direction;
    }
}

File: app/src/main/java/com/dev/nicola/allweather/utils/WeatherUtils.kt
package com.dev.nicola.allweather.utils

import com.dev.nicola.allweather.R

/**
 * Created by Nicola on 17/08/2016.
 */
object WeatherUtils {

    fun getWeatherIcon(condition: String): Int {
        val icon: Int

        when (condition) {
            "1000", "clear-day", "32" -> icon = R.drawable.clear_day
            "clear-night", "31" -> icon = R.drawable.clear_night
            "1003", "partly-cloudy-day", "30", "28" -> icon = R.drawable.partly_cloudy_day
            "partly-cloudy-night", "29", "27" -> icon = R.drawable.partly_cloudy_night
            "1087", "1009", "1006", "cloudy", "26" -> icon = R.drawable.cloud
            "1195", "1192", "1189", "1186", "1183", "1063", "rain", "11", "12" -> icon = R.drawable.rain
            "1225", "1222", "1219", "1216", "1213", "1210", "1114", "1066", "snow", "16" -> icon = R.drawable.snow
            "1072", "1069", "sleet", "18" -> icon = R.drawable.sleet
            "1117", "wind", "24" -> icon = R.drawable.wind
            "1147", "1135", "1030", "fog", "20" -> icon = R.drawable.fog
            "1273", "1276", "1279", "1282", "4", "37", "38", "39", "47" -> icon = R.drawable.storm
            else -> icon = R.drawable.unknown
        }
        return icon
    }


    fun getWindDirection(degrees: Int): String {
        val direction: String
        val cardinal = arrayOf("N", "NNE", "NE", "ENE", "E", "ESE", "SE", "SSE", "S", "SSW", "SW", "WSW", "W", "WNW", "NW", "NNW")
        val n = (degrees / 22.5 + 0.5).toInt()
        direction = cardinal[n % cardinal.size]
        return direction
    }
}
